packages/integrations/solid/src/client.ts

```diff
@@ -1,14 +1,12 @@
+import { Suspense } from 'solid-js';
 import { createComponent, hydrate, render } from 'solid-js/web';
 
 export default (element: HTMLElement) =>
 	(Component: any, props: any, slotted: any, { client }: { client: string }) => {
-		// Prepare global object expected by Solid's hydration logic
-		if (!(window as any)._$HY) {
-			(window as any)._$HY = { events: [], completed: new WeakSet(), r: {} };
-		}
```

Comment:
This code has been moved to the `renderHydrationScript` directive thing and now using the [official hydration script](https://www.solidjs.com/guides/server#hydration-script) generated by Solid's `generateHydrationScript()` function.

Background: The hydration script creates a global `_$HY` object. During Solid's SSR phase, it may serialize data and inject it into inline scripts on the page that depend on the existence of the `_$HY` object. These hydration event scripts look similar to the snippet below:

```html
<script>(self.$R=self.$R||{})["s0"]=[]_$HY.r["s00-2-0"]="$$f"`</script>
```

Hydration events are uncommon in sync SSR, but very common in async SSR (especially when using createResource, Suspense or ErrorBoundary). This file `packages/integrations/solid/src/client.ts` loads too late to handle any actual hydration event scripts. Therefore it becomes necessary to inject the hydration script into HTML directly, before any of the hydration event scripts.